CMakeLists.txt

```diff
@@ -1,59 +1,119 @@
 # Require at least version 2.8.12 of CMake
 cmake_minimum_required(VERSION 2.8.12)
 
+if(CMAKE_VERSION VERSION_EQUAL 3.0 OR CMAKE_VERSION VERSION_GREATER 3.0)
+    cmake_policy(SET CMP0042 NEW)
+endif()
+
 # Set the project name
 project(CoreCLR)
 
-set(CORECLR_SET_RPATH ON)
+# Include cmake functions
+include(functions.cmake)
+
+# Set commonly used directory names
+set(CLR_DIR ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
I do understand that it is a little difficult to see the diff. The reason I did the reordering is so that related code are together and you don't need to scroll up & down to understand what is going on. People have been adding changes in file all over the place. It might be helpful to understand what I am saying if you could look at just my changed file and not the diff. If you want to move any specific piece of code within the block of related code I could do that. If you also wish to move the whole block of related code up and down....I could do that to. But I do not think reverting back to the old way is right to do.